clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3019,4 +3021,18 @@ protected LeaderProducerCallback createProducerCallback(
   protected Lazy<VeniceWriter<byte[], byte[], byte[]>> getVeniceWriter() {
     return veniceWriter;
   }
+
+  @Override
+  protected void maybeUpdateRegionRTOffsetLagCacheAfterReadyToServe(
+      PartitionConsumptionState partitionConsumptionState) {
+    long currTimestamp = System.currentTimeMillis();
+    if ((currTimestamp - partitionConsumptionState.getLatestRTOffsetLagCacheUpdateTimestamp()) > MINUTES.toMillis(1)) {
+      for (String sourceRealTimeTopicKafkaURL: getRealTimeDataSourceKafkaAddress(partitionConsumptionState)) {
+        partitionConsumptionState.updateRegionRTOffsetLagCache(
+            sourceRealTimeTopicKafkaURL,
+            measureRTOffsetLagForSingleRegion(sourceRealTimeTopicKafkaURL, partitionConsumptionState, false));
+      }
+      partitionConsumptionState.setLatestRTOffsetLagCacheUpdateTimestamp(currTimestamp);
+    }
```

Comment:
Hmm... it seems a bit messy to have one timestamp that we can get and set but it represents the entirety of the map having been updated...

I wonder if we could tweak the design of the API to push down the logic into `PCS`. Something like:

```
  void maybeUpdateRegionRTOffsetLagCacheAfterReadyToServe(LagMeasurer lm);
```

Which we call like this:

```
  partitionConsumptionState(srcKafka -> measureRTOffsetLagForSingleRegion(srcKafka, partitionConsumptionState, false));
```

And then inside that function it checks some private state about the last update time, and does the iteration on the map and calls the closure for each key. Something like that.

IDK, what do you think?

Reply:
Sure, exposing less and adding less logic to the SIT classes sounds good. I can update on this.